ts-shared/jl4-lsp-client/custom-protocol.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * Custom extensions to the LSP for JL4
+ *
+ * TODO:
+ * - Look into whether shld add the other viz requests here;
+ *   not sure if that'd make sense since the viz stuff is done through commands
+ *   (as opposed to custom methods).
+ *
+ * Examples of other languages that extend the LSP:
+ * - https://github.com/rust-lang/rust-analyzer/blob/master/editors/code/src/lsp_ext.ts
+ * - https://github.com/Dart-Code/Dart-Code/blob/master/src/shared/analysis/lsp/custom_protocol.ts
+ */
+
+import { RequestType } from 'vscode-languageclient'
```

Comment:
This is changed somewhat in the next PR --- I no longer expose the implementational detail that I'm using this type from some vscode-related package in the next PR.